units/periodic_builds.go

```diff
@@ -96,15 +96,23 @@ func (j *periodicBuildJob) Run(ctx context.Context) {
 		}))
 	}()
 
-	versionID, versionError := j.addVersion(ctx, *definition)
-	if versionError != nil {
-		// if the version fails to be added, create a stub version and
+	mostRecentRevision, err := model.FindLatestRevisionForProject(j.ProjectID)
+	if err != nil {
+		j.AddError(err)
+	}
+	versionID, versionErr := j.addVersion(ctx, *definition, mostRecentRevision)
+
+	if versionErr != nil {
+		// If the version fails to be added, create a stub version and
 		// log an event so users can get notified when notifications are configured
 		metadata := model.VersionMetadata{
 			IsAdHoc:         true,
 			Message:         definition.Message,
 			PeriodicBuildID: definition.ID,
 			Alias:           definition.Alias,
+			Revision: model.Revision{
```

Comment:
If `mostRecentRevision` is empty, the Version created will end up having an empty revision as well. I assume that's fine for stub versions?